controller/src/main/java/org/jboss/as/controller/operations/global/ReadFeatureDescriptionHandler.java

```diff
@@ -392,22 +392,24 @@ private ModelNode describeFeature(final Locale locale, final ImmutableManagement
             String capabilityName = cap.getName();
             if (cap.isDynamicallyNamed()) {
                 PathAddress aliasAddress = createAliasPathAddress(registration, pa);
-                capabilityName = cap.getDynamicName(aliasAddress);
+                if (aliasAddress.size() > 0) {
+                    capabilityName = cap.getDynamicName(createAliasPathAddress(registration, pa));
```

Comment:
Why not re-use aliasAddress like it used to be?

Reply:
Because an empty address creates a NPE in AbstractCapability

Reply:
I don't see how it's different.

Reply:
line 396 should reuse aliasAddress